interpreter/program.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The interpreter package provides functions to evaluate CEL programs against
+// a series of inputs and functions supplied at runtime.
+package interpreter
+
+import (
+	"fmt"
+	"github.com/google/cel-go/common"
+	expr "github.com/google/cel-spec/proto/v1/syntax"
+	"strings"
+)
+
+// Program contains instructions with some metadata and optionally run within
+// a container, e.g. module name or package name.
+type Program interface {
+	// Container is the module or package name where the program is run. The
+	// container is used to resolve type names and identifiers.
+	Container() string
+
+	// Instructions return an InstructionStepper which can be used to iterate
+	// through the program. Each call to Instructions generates a new stepper.
+	Instructions() InstructionStepper
+
+	// Metadata used to determine source locations of sub-expressions.
+	Metadata() Metadata
+}
+
+// IntructionStepper steps through program instructions and provides an option
+// to jump a certain number of instructions forward or back.
+type InstructionStepper interface {
+	// Next returns the next instruction, or false if the end of the program
+	// has been reached.
+	Next() (Instruction, bool)
+
+	// JumpCount moves a relative count of instructions forward or back in the
+	// program and returns whether the jump was successful.
+	//
+	// A jump may be unsuccessful if the number of instructions to jump exceeds
```

Comment:
Comment should start with "JumpCount".

Reply:
It does, this is the second line of comments about when a jump may not be successful.